src/nbt/compound.rs

```diff
@@ -113,3 +112,27 @@ impl NbtCompound {
             .and_then(|tag| tag.extract_long_array())
     }
 }
+
+impl FromIterator<(String, NbtTag)> for NbtCompound {
+    fn from_iter<T: IntoIterator<Item = (String, NbtTag)>>(iter: T) -> Self {
+        Self {
+            child_tags: HashMap::from_iter(iter),
+        }
+    }
+}
+
+impl IntoIterator for NbtCompound {
+    type Item = (String, NbtTag);
+
+    type IntoIter = IntoIter<String, NbtTag>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.child_tags.into_iter()
+    }
+}
+
+impl Extend<(String, NbtTag)> for NbtCompound {
```

Comment:
Change order. Things related to `(String, NbtTag)` should be close together.

Reply:
I put it like this because both FromIterator and IntoIterator are about iterators

Reply:
oh okay